README.md

```diff
@@ -15,11 +15,30 @@ Please ensure all tasks have a description that includes a clear definition of w
 
 Builds are run automatically out of this repository. Basic workflow is:
 
-- Create a branch and implement your changes
-- The Docker image is built on the branch and as part of the pull request
-- Once the PR is merged, a build will be run and pushed to `latest` and `sudo-latest` from master
-- When a new stable image is ready, create a release with a version number like `0.0.6` and
-this will be pushed to `0.0.6` and `sudo-0.0.6`.
+- Create a branch and implement your changes.
+- The Docker image is built on the branch and as part of the pull request.
+- The integration test will be run on PR, merge and release events.
+- Once the PR is merged, a build will run. The resulted image will be tagged with git `head` and `latest` and pushed to ECR.
+- When decide to release a new stable image, create a release with a version number with `major.minor.patch`, e.g.,  `2.0.1`.
+It will trigger the build and image will be tagged with `2.0.1` and `stable` and pushed to ECR.
+
+## Packages' version maintenance and upgrade
+
+The base environment is using conda and docker image is built in two stages:
+
+1. Create conda env and install as many as possible packages from `conda-forge`. Then `pip install` the rest, e.g., most `odc-` packages.
+2. Copy the conda env to a new ubuntu image.
+
+To speed up the building, the build workflow pulls the cache stored on ECR. Though with every build, the cache layers starting from `pip install` will be busted such that the newest versions of `odc-` packages will be installed. Thus, for version upgrade on these packages, creating a release will be enough.
```

Comment:
'busted'? As in they'll be wrong, or they'll get evicted from the cache first?
```suggestion
To speed up the build, the workflow pulls images from a cache stored on ECR. However, with every build the cache layers starting from `pip install` will be discarded, so that the newest versions of `odc-` packages will be installed. Thus, to perform version upgrades on these packages, creating a release is sufficient.
```

Reply:
They'll be deemed wrong (disregarded, ignored) in the building process. What happens to the cached image depends on other factors, not concerned here.